src/app/shared/layout/header.component.ts

```diff
@@ -201,13 +201,17 @@ export class HeaderComponent implements OnInit {
   }
 
   changeRepositoryInPhaseIfValid(repo: Repo, newRepoString: string) {
+    this.phaseService.isChangingRepo.next(true);
+
     this.githubService.isRepositoryPresent(repo.owner, repo.name).subscribe((isValidRepository) => {
       if (!isValidRepository) {
+        this.phaseService.isChangingRepo.next(false);
         throw new Error('Invalid repository name. Please check your organisation and repository name.');
       }
 
       this.switchRepo(repo);
       this.currentRepo = newRepoString;
+      this.phaseService.isChangingRepo.next(false);
     });
   }
```

Comment:
Some considerations, since `githubService` is already present in `phase.service` as well as the nature of this method, it might be a better idea to move this method into `phase.service` instead of keeping it in `header.component`.

Reply:
I agree, but should I extract the method in this PR? I feel like it might be a good idea to do so in a separate PR.

Reply:
Let's do that in another PR

Reply:
Got it, I'll open a PR to do so once this is merged to avoid a conflict